### DISCUSSION TITLE
evid=1 for all observation and dose records

### DISCUSSION THREAD

**Discussion by mahmed452** [metrumresearchgroup/mrgsolve#841] [Q&A]
I created dosing dataset that has only dosing records as follows:
tail(data)
    time amt  ii addl cmt evid                                         regimen Group  ID        WT
695    0 300 336    1   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 398 131.54055
696  672 300 672    4   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 398 131.54055
697    0 300 336    1   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 399  88.58259
698  672 300 672    4   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 399  88.58259
699    0 300 336    1   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 400  49.41403
700  672 300 672    4   1    1 D: 300 q2w for the first month\n then 300mg q4w     D 400  49.41403

then, i did simulations. here is the input and the output of the simulation:
```
sim=mod %>% data_set(data)%>% 
  zero_re() %>% 
  mrgsim(end=4000,recover=c('regimen','Group','WT','evid')) %>%  as_tibble()
head(sim)
# A tibble: 6 x 9
     ID  time    GUT  CENT    CP regimen     Group    WT  evid
  <dbl> <dbl>  <dbl> <dbl> <dbl> <chr>       <chr> <dbl> <dbl>
1     1     0   0       0   0    A: 300 q 4w A      130.     1
2     1     0 300       0   0    A: 300 q 4w A      130.     1
3     1     1  77.8   219.  3.31 A: 300 q 4w A      130.     1
4     1     2  20.2   270.  4.09 A: 300 q 4w A      130.     1
5     1     3   5.23  277.  4.20 A: 300 q 4w A      130.     1
6     1     4   1.35  274.  4.15 A: 300 q 4w A      130.     1
```
my question is how do i make evid or amt be only 1 or >0 at the dosing records only? in other words, I want the output like this
```
# A tibble: 6 x 9
     ID  time    GUT  CENT    CP regimen     Group    WT  evid
  <dbl> <dbl>  <dbl> <dbl> <dbl> <chr>       <chr> <dbl> <dbl>
1     1     0   0       0   0    A: 300 q 4w A      130.     **0**
2     1     0 300       0   0    A: 300 q 4w A      130.     **1**
3     1     1  77.8   219.  3.31 A: 300 q 4w A      130.     **0**
4     1     2  20.2   270.  4.09 A: 300 q 4w A      130.     **0**
5     1     3   5.23  277.  4.20 A: 300 q 4w A      130.     **0**
6     1     4   1.35  274.  4.15 A: 300 q 4w A      130.     **0**
```

**Comment by kylebaron** [comment]
Hi @mahmed452  - 

You can get that output by using `carry_out` rather than `recover` for evid

```r
sim=mod %>% data_set(data)%>% 
  zero_re() %>% 
  mrgsim(end=4000,recover=c('regimen','Group','WT'), carry_out = "evid") %>%  as_tibble()
head(sim)
```

`recover` is a new(er) feature and I didn't think about this happening. I think this is what should be happening given the current implementation, but needs better documentation around this. 

Please let me know if moving `evid` to `carry_out` fixes it for you.

Thanks,
Kyle

**Comment by mahmed452** [comment]
It worked. Thank you
